age/src/scrypt.rs

```diff
@@ -84,7 +84,7 @@ impl crate::Recipient for Recipient {
         let mut salt = [0; SALT_LEN];
         OsRng.fill_bytes(&mut salt);
 
-        let mut inner_salt = vec![];
+        let mut inner_salt = Vec::with_capacity(SCRYPT_SALT_LABEL.len() + SALT_LEN);
         inner_salt.extend_from_slice(SCRYPT_SALT_LABEL);
         inner_salt.extend_from_slice(&salt);
```

Comment:
This is only 44 bytes, so we should just construct it on the stack:
```suggestion
        let mut inner_salt = [0; SCRYPT_SALT_LABEL.len() + SALT_LEN];
        inner_salt[..SCRYPT_SALT_LABEL.len()].copy_from_slice(SCRYPT_SALT_LABEL);
        inner_salt[SCRYPT_SALT_LABEL.len()..].copy_from_slice(&salt);
```